### ISSUE TITLE
Tesla instead of HTTPotion?

### ISSUE THREAD

**Issue by oleks** [radar/elastic#36]
I noticed that HTTPotion [is deprecated](https://hex.pm/packages/httpotion).. The suggestion from the package author there is to use [Tesla](https://github.com/teamon/tesla) instead.

Changing this is a pretty big change, so I wanted to ask before going ahead with this, whether you have better plans in store?

**Comment by radar** [comment]
@oleks I really don't use this library any more as I am now contracting as a Ruby developer. I have used it plenty of times in the past.

I think if the top-level API that Elastic exposes changes as little as possible then I think it would be fine to replace it with Tesla. I'm guessing a good reason to do that would be so that when you do a `mix deps.get` you have _one_ HTTP library downloaded / compiled instead of two.

**Comment by oleks** [comment]
> @oleks I really don't use this library any more as I am now contracting as a Ruby developer. I have used it plenty of times in the past.

I can take over maintenance of the library for now, if you like?

**Comment by radar** [comment]
Yes please :) Help will be very welcome. I've sent an invite to you just now.
